### PR TITLE
Create a new Question 3 for Part Year Tax Claim

### PR DESCRIPTION
**Pull request by ikennaokpala**
Trello cards:
https://trello.com/c/eXfGf5lL/171-child-benefit-tax-calculator-add-question-part-year-and-complete-logic-for-answering-no

https://trello.com/c/KbE3axrr/172-child-benefit-tax-calculator-part-year-all-children-add-logic-for-answering-yes

This is following the work of @leenagupte in PR #141 
Merge #141 first before merging this PR.
This supersedes #142 

## Factcheck

[Preview link](https://calculators-pr-142.herokuapp.com/child-benefit-tax-calculator/main)
[Preview link for PR 141](https://calculators-pr-141.herokuapp.com/child-benefit-tax-calculator/main)
[GOV.UK](https://gov.uk/child-benefit-tax-calculator/main)

## Expected changes 

Create a new Question 3 that asks if any of the children are only being claimed for a part of the tax year.

#### GOV.UK

screen shot 2016-06-13 at 17 03 53

#### PR 141
screen shot 2016-06-13 at 17 01 57

### After

screen shot 2016-06-13 at 17 00 43

### CODE REVIEW

**Comment by leenagupte** [comment]
@ikennaokpala The PR looks good, but you probably want someone who is more comfortable with javascript to review the toggling. 

@pmanrubia Are you free to review the toggling function?

File: spec/models/child_benefit_tax_calculator_spec.rb [hunk 1]
```diff
@@ -764,6 +783,33 @@
         )
         expect(calc.benefits_claimed_amount.round(2)).to eq(621.0)
       end
+
+      it "should set the start date to start of the selected tax year" do
```
Comment (pmanrubia) [inline spec/models/child_benefit_tax_calculator_spec.rb]: This test does not seem to be related to the validation `#is_part_year_claim`. If these are additional tests, would you mind creating a separate commit explaining the cases you are covering?

The same applies to the one below:
```ruby
it "should set the stop date to end of the selected tax year
```

File: app/models/child_benefit_tax_calculator.rb [hunk 2]
```diff
@@ -46,7 +48,11 @@ def nothing_owed?
   end
 
   def has_errors?
-    errors.any? || starting_children.select { |c| c.errors.any? }.any?
+    errors.any? || starting_children_errors?
+  end
+
+  def starting_children_errors?
```
Comment (pmanrubia) [inline app/models/child_benefit_tax_calculator.rb]: Not sure about this. If `is_part_year_claim=='no'` then there are no errors?. I might be missing something. My understanding is that we would need to validate all the children regardless the `is_part_year_claim` value.

File: spec/models/child_benefit_tax_calculator_spec.rb [hunk 3]
```diff
@@ -121,6 +121,25 @@
         ).has_errors?).to eq(false)
       end
     end
+
+    describe "#is_part_year_claim" do
```
Comment (pmanrubia) [inline spec/models/child_benefit_tax_calculator_spec.rb]: I find it a bit confusing that for validating the `:is_part_year_claim` we need to setup the attributes: `:children_count` and `:year`

Would something like this work for you:
```ruby
it "should not contain errors if tax claim duration is set to no" do
  calc = ChildBenefitTaxCalculator.new(is_part_year_claim: "no")
  calc.valid?

  expect(calc.errors).to_not have_key(:is_part_year_claim)
end
```

File: app/models/child_benefit_tax_calculator.rb (hunk 2 above)
Comment (leenagupte) [inline app/models/child_benefit_tax_calculator.rb]: @pmanrubia If `is_part_year_claim == 'no'` then the start and end dates for the child should default to the start and end dates for the tax year selected. User doesn't input any dates so there is no need for validation.

Comment (pmanrubia) [inline app/models/child_benefit_tax_calculator.rb]: @leenagupte: we cannot assume that `is_part_year_claim=='no'` implies that I am not going to run the validations of the children. I would expect any new validation added to the children to run in all situations.

**Comment by pmanrubia** [comment]
@leenagupte the JS seems to be consistent with previous code. It might need further refactoring once the logic is fully implemented with the following cards.

**Comment by leenagupte** [comment]
@pmanrubia Thanks.

**Comment by pmanrubia** [comment]
Superseded by #146
